osu.Game.Tests/Visual/Gameplay/TestScenePauseInputHandling.cs

```diff
@@ -325,32 +372,40 @@ public void TestOsuHitCircleNotReceivingInputOnResume()
         public void TestOsuHitCircleNotReceivingInputOnResume_PauseWhileHoldingSameKey()
         {
             KeyCounter counter = null!;
+            int leftFrameCount = 0;
 
             loadPlayer(() => new OsuRuleset());
             AddStep("get key counter", () => counter = this.ChildrenOfType<KeyCounter>().Single(k => k.Trigger is KeyCounterActionTrigger<OsuAction> actionTrigger && actionTrigger.Action == OsuAction.LeftButton));
 
             AddStep("press Z", () => InputManager.PressKey(Key.Z));
             AddAssert("circle hit", () => Player.ScoreProcessor.HighestCombo.Value, () => Is.EqualTo(1));
-            assertCountOfMatchingReplayFrames<OsuReplayFrame>("left button pressed once in replay", f => f.Actions.Contains(OsuAction.LeftButton), 1);
+            AddUntilStep("left button pressed at least once in replay",
+                () => Player.Score.Replay.Frames.OfType<OsuReplayFrame>().Count(f => f.Actions.Contains(OsuAction.LeftButton)),
+                () => Is.GreaterThan(0));
 
+            seekTo(50);
             AddStep("pause", () => Player.Pause());
             AddStep("release Z", () => InputManager.ReleaseKey(Key.Z));
+            AddStep("store count of left presses in replay", () => leftFrameCount = Player.Score.Replay.Frames.OfType<OsuReplayFrame>().Count(f => f.Actions.Contains(OsuAction.LeftButton)));
 
             AddStep("resume", () => Player.Resume());
             AddStep("go to resume cursor", () => InputManager.MoveMouseTo(this.ChildrenOfType<OsuResumeOverlay.OsuClickToResumeCursor>().Single()));
-            AddStep("press Z to resume", () => InputManager.PressKey(Key.Z));
-            AddStep("release Z", () => InputManager.ReleaseKey(Key.Z));
+            AddStep("press and release Z to resume", () => InputManager.Key(Key.Z));
 
             checkKeyCounterState(() => counter, 1, false);
-            assertCountOfMatchingReplayFrames<OsuReplayFrame>("left button pressed once in replay", f => f.Actions.Contains(OsuAction.LeftButton), 1);
+            AddUntilStep("left button not pressed again in replay",
+                () => Player.Score.Replay.Frames.OfType<OsuReplayFrame>().Count(f => f.Actions.Contains(OsuAction.LeftButton)),
+                () => Is.Not.GreaterThan(leftFrameCount));
```

Comment:
The latest change made the test very hard to comprehend. Instead of tracking how many times the button was pressed, this is now tracking how *many* frames exist for the press of a button. This feels quite off.

Can this be swapped with a method that goes through all frames each time and counts how many times the button is pressed after a previous release?

Reply:
I don't understand both the concern and what you want me to do here.

The reason that this is "counting frames" is that there is an *indeterminate* number of frames in this test scenario that may contain a pressed button after a press. It may be one, it may be more than one. It's not like the key counter where the counter increments by one every time you press.